metron-platform/metron-common/src/main/java/org/apache/metron/common/utils/KafkaUtils.java

```diff
@@ -68,12 +68,14 @@
     return ret;
   }
 
-  public List<String> fromEndpoint(String url) throws URISyntaxException {
+  public List<String> fromEndpoint(String url){
     List<String> ret = new ArrayList<>();
     if(url != null) {
-      URI uri = new URI(url);
-      int port = uri.getPort();
-      ret.add(uri.getHost() + ((port > 0)?(":" + port):""));
+      Iterable<String> splits = Splitter.on("//").split(url);
+      if(Iterables.size(splits) == 2) {
+        String hostPort = Iterables.getLast(splits);
+        ret.add(hostPort);
```

Comment:
Hi @justinleet , perhaps it is because I don't know how SASL_PLAINTEXT endpoints work, but I don't get the change above. I fully understand the deleted lines, and don't see how the substituted lines are equivalent, or what was wrong with the former.  Would you mind giving a bit of explanation?  Thanks.

Also, I just went back and re-read your introduction to this PR, and followed the link to (5d36c79)[https://github.com/apache/metron/commit/5d36c79effec75f6ac95fa587f80da0bd5420135].  It appears you've exactly reversed a chunk of @cestella 's patch of Mar 23.  Was this intended?  I'm even more confused :-)